src/components/MultiHopTrade/components/TradeInput/components/TradeQuotes/TradeQuotes.tsx

```diff
@@ -93,14 +156,71 @@ export const TradeQuotes: React.FC<TradeQuotesProps> = memo(({ isLoading, onBack
     isTradeQuoteRequestAborted,
     bestQuoteData?.quote,
     bestQuoteData?.inputOutputRatio,
-    tradeQuoteDisplayCache,
+    validTradeQuotesDisplayCache,
     activeQuoteMeta,
     isTradeQuoteApiQueryPending,
     isLoading,
     isSwapperQuoteAvailable,
     onBack,
   ])
 
+  const invalidQuotes = useMemo(() => {
+    if (isTradeQuoteRequestAborted) {
+      return []
+    }
+
+    const bestTotalReceiveAmountCryptoPrecision = bestQuoteData?.quote
+      ? getBuyAmountAfterFeesCryptoPrecision({
+          quote: bestQuoteData.quote,
+        })
+      : undefined
+
+    return invalidTradeQuotesDisplayCache.map((quoteData, i) => {
+      const { swapperName, id, errors, isStale } = quoteData
+
+      const isActive = activeQuoteMeta !== undefined && activeQuoteMeta.identifier === id
+
+      return (
+        <TradeQuote
+          isActive={isActive}
+          isLoading={Boolean(isTradeQuoteApiQueryPending[quoteData.swapperName]) && isStale}
+          isRefetching={
+            isLoading ||
+            isTradeQuoteApiQueryPending[quoteData.swapperName] ||
+            !isSwapperQuoteAvailable[swapperName]
+          }
+          isBest={i === 0 && errors.length === 0}
+          key={id}
+          quoteData={quoteData}
+          bestTotalReceiveAmountCryptoPrecision={bestTotalReceiveAmountCryptoPrecision}
+          bestInputOutputRatio={bestQuoteData?.inputOutputRatio}
+          onBack={onBack}
+        />
+      )
+    })
+  }, [
+    isTradeQuoteRequestAborted,
+    bestQuoteData?.quote,
+    bestQuoteData?.inputOutputRatio,
+    invalidTradeQuotesDisplayCache,
+    activeQuoteMeta,
+    isTradeQuoteApiQueryPending,
+    isLoading,
+    isSwapperQuoteAvailable,
+    onBack,
+  ])
+
+  const invalidQuotesLength = useMemo(
+    () =>
+      invalidTradeQuotesDisplayCache.filter(
+        invalidQuote =>
+          ![TradeQuoteValidationError.UnknownError, SwapperTradeQuoteError.UnknownError].includes(
+            invalidQuote.errors?.[0]?.error,
```

Comment:
preferably blocking: we should introspect all errors in the array, as opposed to element `[0]`

Reply:
Is there any reasons why we should do this? 

We are displaying the first error in the tag, the filter should be based on the first error or the user won't understand anything?

Reply:
See https://github.com/shapeshift/web/pull/8087/files#r1833445284 🙏